src/System.Runtime.Extensions/src/System/IO/Path.Windows.cs

```diff
@@ -41,145 +41,58 @@ public static string GetFullPath(string path)
             {
-                startIndex += PathInternal.ExtendedPathPrefix.Length;
+                // We can't really know what is valid for all cases of extended paths.
+                //
+                //  - object names can include other characters as well (':', '/', etc.)
+                //  - even file objects have different rules (pipe names can contain most characters)
+                //
+                // As such we will do no further analysis of extended paths to avoid blocking known and unknown
+                // scenarios as well as minimizing compat breaks should we block now and need to unblock later.
+                return path;
             }
 
-            // Move past the colon
-            startIndex += 2;
-
-            if ((path.Length > 0 && path[0] == VolumeSeparatorChar)
-                || (path.Length >= startIndex && path[startIndex - 1] == VolumeSeparatorChar && !PathInternal.IsValidDriveChar(path[startIndex - 2]))
-                || (path.Length > startIndex && path.IndexOf(VolumeSeparatorChar, startIndex) != -1))
+            bool isDevice = PathInternal.IsDevice(path);
+            if (!isDevice)
             {
-                throw new NotSupportedException(SR.Argument_PathFormatNotSupported);
-            }
+                // Toss out paths with colons that aren't a valid drive specifier.
+                // Cannot start with a colon and can only be of the form "C:".
+                // (Note that we used to explicitly check "http:" and "file:"- these are caught by this check now.)
+                int startIndex = PathInternal.PathStartSkip(path);
 
-            if (isExtended)
-            {
-                // If the path is in extended syntax, we don't need to normalize, but we still do some basic validity checks
-                if (!ValidateExtendedPath(path))
+                // Move past the colon
+                startIndex += 2;
+
+                if ((path.Length > 0 && path[0] == VolumeSeparatorChar)
+                    || (path.Length >= startIndex && path[startIndex - 1] == VolumeSeparatorChar && !PathInternal.IsValidDriveChar(path[startIndex - 2]))
+                    || (path.Length > startIndex && path.IndexOf(VolumeSeparatorChar, startIndex) != -1))
                 {
-                    throw new ArgumentException(SR.Arg_PathIllegal);
+                    throw new NotSupportedException(SR.Argument_PathFormatNotSupported);
                 }
+            }
 
-                // \\?\GLOBALROOT gives access to devices out of the scope of the current user, we
-                // don't want to allow this for security reasons.
-                // https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247.aspx#nt_namespaces
-                if (path.StartsWith(@"\\?\globalroot", StringComparison.OrdinalIgnoreCase))
-                    throw new ArgumentException(SR.Arg_PathGlobalRoot);
-
+            // Technically this doesn't matter but we used to throw for this case
```

Comment:
This is what is in the desktop currently. Ironically it doesn't catch any of the other ways to do this. `\\.\globalroot`, or `\\.\global\globalroot`, etc., etc...